src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/RegistrationDataEntry.component.js

```diff
@@ -1,30 +1,74 @@
 // @flow
 import React from 'react';
+import i18n from '@dhis2/d2-i18n';
 import type { OwnProps } from './RegistrationDataEntry.types';
 import { EnrollmentRegistrationEntry, TeiRegistrationEntry } from '../../../DataEntries';
 import { scopeTypes } from '../../../../metaData';
 import { useScopeInfo } from '../../../../hooks/useScopeInfo';
+import { useRegistrationFormInfoForSelectedScope } from '../../../DataEntries/common/useRegistrationFormInfoForSelectedScope';
+import { InfoIconText } from '../../../InfoIconText';
+import { useCurrentOrgUnitInfo } from '../../../../hooks/useCurrentOrgUnitInfo';
+
+
+const translatedTextWithStylesForProgram = (trackedEntityName, programName, orgUnitName) =>
+    (<>
+        {i18n.t('Saving a {{trackedEntityName}} in', { trackedEntityName })} <b>{programName}</b>
+        {orgUnitName && <>{' '}{i18n.t('in')} <b>{orgUnitName}</b></>}.
+    </>);
+
+
+const translatedTextWithStylesForTei = (trackedEntityName, orgUnitName) =>
+    (<>
+        {i18n.t('Saving a {{trackedEntityName}}', { trackedEntityName })} <b>{i18n.t('without')}</b> {i18n.t('enrollment')}
+        {orgUnitName && <>{' '}{i18n.t('in')} <b>{orgUnitName}</b></>}.{' '}
+        {i18n.t('Enroll in a program by selecting a program from the top bar.')}
+    </>);
 
 
 export const RegistrationDataEntry = ({ selectedScopeId, dataEntryId }: OwnProps) => {
-    const { scopeType } = useScopeInfo(selectedScopeId);
+    const { scopeType, trackedEntityName, programName } = useScopeInfo(selectedScopeId);
+    const { registrationMetaData } = useRegistrationFormInfoForSelectedScope(selectedScopeId);
+    const orgUnit = useCurrentOrgUnitInfo();
 
     return (
         <>
             {
                 scopeType === scopeTypes.TRACKER_PROGRAM &&
-                <EnrollmentRegistrationEntry
-                    selectedScopeId={selectedScopeId}
-                    id={dataEntryId}
-                />
+                <>
+                    <EnrollmentRegistrationEntry
+                        id={dataEntryId}
+                        selectedScopeId={selectedScopeId}
+                        enrollmentMetadata={registrationMetaData}
+                        saveButtonText={'Save new'}
+                        onSave={() => alert('onSave will save in the future')}
+                        onGetUnsavedAttributeValues={() => console.log('onGetUnsavedAttributeValues will be here in the future in the future')}
+                        onPostProcessErrorMessage={() => console.log('onPostProcessErrorMessage will be here in the future in the future')}
+                        onUpdateField={() => console.log('onUpdateField will be here in the future in the future')}
+                        onStartAsyncUpdateField={() => console.log('onStartAsyncUpdateField will be here in the future in the future')}
+                    />
```

Comment:
Here I would say you will see the benefit of creating the `EnrollmentRegistrationEntry` and `TeiRegistrationEntry ` components.